bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -178,29 +182,37 @@ void ReplicaImp::messageHandler(MessageBase *msg) {
       }
     }
   }
-
-  if (!isCollectingState()) {
-    // The message validation of few messages require time-consuming processes like
-    // digest calculation, signature verification etc. Such messages are identified
-    // by review. During the review we also check if asynchronous message validation
-    // is feasible or not for the given message. After identification, true returning
-    // shouldValidateAsync() member function is added to these message which indicates
-    // that Asynchronous validation is possible for these messages.
-    // In this function we check shouldValidateAsync() and decide if asynchronous
-    // validation will happen or synchronous validation will happen.
-    // prePrepareFinalizeAsyncEnabled flag will be removed in 1.7, when this entire
-    // async behaviour will be assumed to be default for the chosen messages.
-    if (getReplicaConfig().prePrepareFinalizeAsyncEnabled && trueTypeObj->shouldValidateAsync()) {
-      asyncValidateMessage<T>(trueTypeObj);
-      return;
-    } else {
-      if (validateMessage(trueTypeObj)) {
-        onMessage<T>(trueTypeObj);
+  if constexpr (std::is_same_v<T, StateTransferMsg>) {
+    if (validateMessage(trueTypeObj))
+      onMessage<T>(trueTypeObj);
+    else
+      delete trueTypeObj;
+    return;
+  }
+  if (!isCollectingState() && validateMessage(trueTypeObj)) {
+    if (!isCollectingState()) {
+      // The message validation of few messages require time-consuming processes like
+      // digest calculation, signature verification etc. Such messages are identified
+      // by review. During the review we also check if asynchronous message validation
+      // is feasible or not for the given message. After identification, true returning
+      // shouldValidateAsync() member function is added to these message which indicates
+      // that Asynchronous validation is possible for these messages.
+      // In this function we check shouldValidateAsync() and decide if asynchronous
+      // validation will happen or synchronous validation will happen.
+      // prePrepareFinalizeAsyncEnabled flag will be removed in 1.7, when this entire
+      // async behaviour will be assumed to be default for the chosen messages.
```

Comment:
We support 120 chars per line now, it is worth using them to have a comment shorter.

Reply:
This is not related to me.